### PR TITLE
Interpreter: avoid member_offset_iterator

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

File: src/goto-programs/interpreter_evaluate.cpp [hunk 1]
```diff
@@ -955,16 +913,40 @@ void interpretert::evaluate(
     }
     else if(!address.is_zero())
     {
+      if(expr.id()==ID_byte_extract_little_endian ||
+         expr.id()==ID_byte_extract_big_endian)
+      {
+        // extract_from will be non-empty, otherwise evaluate_address would have
```
Comment (smowton) [inline src/goto-programs/interpreter_evaluate.cpp]: In that case `INVARIANT` it

**Review by smowton — COMMENTED** [review]
Could you double check if there any tests exercising this code? The change looks good however.

**Comment by smowton** [comment]
@thk123 flagging this for a test-gen bump.

**Comment by thk123** [comment]
Created diffblue/test-gen#1793 - but this probably needs a rebase

**Comment by peterschrammel** [comment]
Rebased

**Comment by thk123** [comment]
Seems to be failing on TG - will investigate today.

**Comment by tautschnig** [comment]
@thk123 I have turned a comment into an `INVARIANT` as @smowton suggested. Let me know if there is anything I can do to help your investigation!

**Comment by thk123** [comment]
Re-running now.

**Comment by peterschrammel** [comment]
Rebased this to make it testable with TG.

**Comment by thk123** [comment]
TG pointer updated.

**Comment by thk123** [comment]
TG currently failing - will investigate.

**Comment by tautschnig** [comment]
@allredj This is a pre-existing PR - does it now also get auto-test-gen-tested?
@peterschrammel In particular if the answer to the above is "no": I have removed the second commit from this PR, and would hope that test-gen now passes.

**Comment by allredj** [comment]
@tautschnig it should. I'll keep you posted.

**Comment by allredj** [comment]
@peterschrammel @tautschnig actually I just checked the previous failure, and it happened in a test that is not yet part of the check. So the PR on test-gen will have to be updated manually.

**Comment by peterschrammel** [comment]
Rebased the bump

**Review by peterschrammel — APPROVED** [review]
TG bump passes

**Pull request merged**